certificate-renewal/Certificate Renewal JSON contract.md

```diff
@@ -0,0 +1,173 @@
+	 - 1-**shareLoanId**: 4-character share or loan ID
+     - 2-**shareLoanName**: share nickname if available otherwise the share description
+	 - 3-**shareLoanBalance**: the current share available balance
+
+
+## Request (PROCESSDATA):
+```json
+{
+  "rgState": "PROCESSDATA",
+  "powerOnFileName": "BANNO.CD.RENEW.V1.POW",
+  "userChrList":[
+    {"id": 1, "value": "0123456789S0123"},
+    {"id": 2, "value": ""},
+    {"id": 3, "value": ""},
+    {"id": 4, "value": ""},
+    {"id": 5, "value": ""}
+  ],
+  "userNumList": [
+    {"id": 1, "value": 1},
+    {"id": 2, "value": 123456.78},
+    {"id": 3, "value": null},	
+    {"id": 4, "value": null},	
+    {"id": 5, "value": null}	
+],
+  "rgSession": 1
+}
+```
+ - userNumList[1]: User selected option
+	 - Represents the CD renewal option the member selected.
+ - userNumList[2]: User selected option
+	 - The transfer in/out amount
+ - userCharList[1]: 10-digit account number, 'S' or 'L' and 4-digit share/Loan ID
+	 - Represents the source or target share/loan of the transfer
+
+## Successful Response:
+```json
+{
+  "results":  "success",
+    "updatedState": {
+      "name": "My Wonderful CD",
+      "maturityDate": "01/01/21",
+      "newStatus": "Renew",
+      "transferSLId": "1234567890S0010",
+      "amount": "12,345.67"
+    }
+  }
+```
+ - ***updatedState***
+  	 - 1-**name**: Share nickname of targeted certificate/club share. If no nickname, share description.
+  	 - 2-**maturityDate**: The current maturity date of the targeted certificate/club.
+  	 - 3-**minimumBalance**: The current minimum balance requirement of targeted certificate/club.
+  	 - 4-**newStatus**: The new maturity option set for the targeted certificate/club.
+  	 - 5-**transferSLID**: If the current maturity option includes transferring funds into or
+  	 - out of the CD, this will be the current 10-digit account number and share/loan ID.
+  	 - 6-**transferAmount**: If the current maturity option includes transferring funds into or
+  	 - out of the CD, this will be the current amount to be transferred.
+
+
+## Unsuccessful Response:
+If any request is not successful for any number of reasons, or the desired action results in an error condition, the PowerOn will respond with the following error structure :
+
+```json
+{
+  "errorCode": XXX,
+  "loggingErrorMessage": "[error code detail]"
+}
+```
+## Error Codes
+
+| errorCode   | loggingErrorMessage                                             |
+|-------------|-----------------------------------------------------------------|
+| 500         | Generic Error                                                   |
+| 501         | Configuration file error (error opening, reading or processing) |
+| 502         | Invalid account or share (based on account type/share warning)  |
+| 503         | Maturity selection previously made - contact CU                 |
+| 504         | Processing Error                                                |
+
```

Comment:
We'll need to do the same cross-account blocking and error responses we did for wd-by-check.

Reply:
done - next commit